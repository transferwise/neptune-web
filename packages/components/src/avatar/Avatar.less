@import (reference) '@transferwise/neptune-css/src/variables/neptune-tokens.less';

@avatar-diameter-lg: 72px;
@avatar-diameter-md: 48px;
@avatar-diameter-sm: 24px;

.tw-avatar {
  position: relative;
  border-radius: 50%;
  user-select: none;
  box-sizing: border-box;
  cursor: default;

  & &__content {
    display: flex;
    justify-content: center;
    align-items: center;
    border-radius: 50%;
    overflow: hidden;
    background-color: var(--color-background-neutral);
    height: 100%;
    width: 100%;
    max-height: 100%;
    max-width: 100%;
  }

  &--outlined {
    border: 1px solid var(--color-control-accent);
    cursor: pointer;

    &:hover {
      border-color: var(--color-control-accent-hover);
    }
  }

  &--dark &__content {
    background: @color-base-navy-mid;
    color: @color-text-control;
  }

  &--thumbnail {
    & * {
      width: 100%;
      height: 100%;
      object-fit: cover;
    }
  }

  &--icon {
    & * {
      max-height: 100%;
      max-width: 100%;
    }
  }

  &.tw-avatar--initials,
  &.tw-avatar--icon {
    color: @color-text-primary;
  }

  &.tw-avatar--initials,
  &.tw-avatar--emoji {
    // There is a problem how emojis & text vertically aligned between Windows and Mac. Setting font family to Arial seems to resolve it.
    font-family: Arial, sans-serif;
  }

  &--sm {
    .avatar-dimension(@avatar-diameter-sm, 1px);
  }

  &--md {
    .avatar-dimension(@avatar-diameter-md, 2px);
  }

  &--lg {
    .avatar-dimension(@avatar-diameter-lg, 3px);
  }

  img {
    width: 100%;
    height: 100%;
  }
}

.avatar-dimension(@avatar-size, @mask-size) {
  min-width: @avatar-size;
  width: @avatar-size;
  height: @avatar-size;

  &.tw-avatar--emoji,
  &.tw-avatar--icon {
    font-size: @avatar-size * 0.5;
  }

  &.tw-avatar--initials {
    font-size: @avatar-size * 0.35;
    font-weight: bold;
  }

  &.tw-avatar--outlined {
    padding: @mask-size;

    &.tw-avatar--thumbnail::after {
      top: @mask-size;
      left: @mask-size;
      right: @mask-size;
      bottom: @mask-size;
    }
  }

  &.tw-avatar--thumbnail::after {
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    border: 1px solid rgba(0, 0, 0, 0.08);
    content: '';
    border-radius: 50%;
  }

  &.tw-avatar--dark.tw-avatar--thumbnail::after {
    border-color: rgba(255, 255, 255, 0.08);
  }
}
