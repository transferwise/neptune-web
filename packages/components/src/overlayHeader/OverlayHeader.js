import React from 'react';
import Types from 'prop-types';

import Header from '../header';
import CloseButton from '../common/closeButton';

import './OverlayHeader.css';

const OverlayHeader = ({ avatar, onClose, logo }) => {
  const closeButton = onClose && <CloseButton onClick={onClose} />;
  return (
    <Header
      className="np-overlay-header p-a-3"
      leftContent={logo}
      rightContent={
        <>
          {avatar}
          {avatar && closeButton && <span className="separator" />}
          {closeButton}
        </>
      }
    />
  );
};

OverlayHeader.defaultProps = {
  avatar: null,
  onClose: null,
};

OverlayHeader.propTypes = {
  /** An Avatar */
  avatar: Types.element,
  logo: Types.node.isRequired,
  /** Function called when the close is clicked */
  onClose: Types.func,
};

export default OverlayHeader;
