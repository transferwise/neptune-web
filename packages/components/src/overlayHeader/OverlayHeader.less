@import (reference) '@transferwise/neptune-css/src/less/addons/_spacing-utilities.less';

.np-overlay-header {
  // This is the same height of FlowNavigation.
  min-height: 97px;
  .separator {
    .m-x-2;
    height: var(--size-32);
    width: 1px;
    background-color: var(--color-border-neutral);
  }
}
