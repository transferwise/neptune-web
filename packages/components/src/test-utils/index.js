import React from 'react';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Provider } from '..';
import { DEFAULT_LOCALE } from '../common/locale';
import en from '../../i18n/en.json';

/**
 * Custom `render` function which wraps passed elements in Provider component
 * Source: https://testing-library.com/docs/example-react-intl/#creating-a-custom-render-function
 */
function customRender(ui, { locale = DEFAULT_LOCALE, messages = en, ...renderOptions } = {}) {
  // eslint-disable-next-line react/prop-types
  function Wrapper({ children }) {
    return <Provider i18n={{ locale, messages }}>{children}</Provider>;
  }
  return render(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from '@testing-library/react';
export { userEvent };
export { customRender as render };
