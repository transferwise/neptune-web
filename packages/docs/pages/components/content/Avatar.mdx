import { LiveEditorBlock, GeneratePropsTable } from '../../../utils';
import { Avatar } from '@transferwise/components';
import { Profile } from '@transferwise/icons';
import code from '../../../liveEditorCode/avatar.code';

<LiveEditorBlock code={code} scope={{ Avatar }} />
<GeneratePropsTable componentName="Avatar" />

<table className="table table-condensed docs-table m-t-5">
  <thead>
    <tr>
      <th>Avatar type</th>
      <th>Avatar size</th>
      <th>Default</th>
      <th>Outline</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Avatar.Type.THUMBNAIL</td>
      <td>Avatar.Type.LARGE</td>
      <td>
        <Avatar type={Avatar.Type.THUMBNAIL} size={Avatar.Size.LARGE}>
          <img
            src="https://transferwise.com/public-resources/assets/flags/square/gbp.svg"
            alt="UK Flag"
          />
        </Avatar>
      </td>
      <td>
        <Avatar type={Avatar.Type.THUMBNAIL} size={Avatar.Size.LARGE} outlined>
          <img
            src="https://transferwise.com/public-resources/assets/flags/square/gbp.svg"
            alt="UK Flag"
          />
        </Avatar>
      </td>
    </tr>
    <tr>
      <td>Avatar.Type.THUMBNAIL</td>
      <td>Avatar.Type.MEDIUM</td>
      <td>
        <Avatar type={Avatar.Type.THUMBNAIL} size={Avatar.Size.MEDIUM}>
          <img
            src="https://transferwise.com/public-resources/assets/flags/square/gbp.svg"
            alt="UK Flag"
          />
        </Avatar>
      </td>
      <td>
        <Avatar type={Avatar.Type.THUMBNAIL} size={Avatar.Size.MEDIUM} outlined>
          <img
            src="https://transferwise.com/public-resources/assets/flags/square/gbp.svg"
            alt="UK Flag"
          />
        </Avatar>
      </td>
    </tr>
    <tr>
      <td>Avatar.Type.THUMBNAIL</td>
      <td>Avatar.Type.SMALL</td>
      <td>
        <Avatar type={Avatar.Type.THUMBNAIL} size={Avatar.Size.SMALL}>
          <img
            src="https://transferwise.com/public-resources/assets/flags/square/gbp.svg"
            alt="UK Flag"
          />
        </Avatar>
      </td>
      <td>
        <Avatar type={Avatar.Type.THUMBNAIL} size={Avatar.Size.SMALL} outlined>
          <img
            src="https://transferwise.com/public-resources/assets/flags/square/gbp.svg"
            alt="UK Flag"
          />
        </Avatar>
      </td>
    </tr>
    <tr>
      <td>Avatar.Type.EMOJI</td>
      <td>Avatar.Type.LARGE</td>
      <td>
        <Avatar size={Avatar.Size.LARGE} type={Avatar.Type.EMOJI} backgroundColor="#D6F5E3">
          <img
            style={{ width: 44, height: 44 }}
            src="https://transferwise.com/public-resources/assets/balances/savings-emoji/google/ðŸŒ±.png"
            alt=""
          />
        </Avatar>
      </td>
      <td>
        <Avatar
          size={Avatar.Size.LARGE}
          type={Avatar.Type.EMOJI}
          backgroundColor="#D6F5E3"
          outlined
        >
          <img
            style={{ width: 44, height: 44 }}
            src="https://transferwise.com/public-resources/assets/balances/savings-emoji/google/ðŸŒ±.png"
            alt=""
          />
        </Avatar>
      </td>
    </tr>
    <tr>
      <td>Avatar.Type.EMOJI</td>
      <td>Avatar.Type.MEDIUM</td>
      <td>
        <Avatar size={Avatar.Size.MEDIUM} type={Avatar.Type.EMOJI} backgroundColor="#D6F5E3">
          <img
            style={{ width: 30, height: 30 }}
            src="https://transferwise.com/public-resources/assets/balances/savings-emoji/google/ðŸŒ±.png"
            alt=""
          />
        </Avatar>
      </td>
      <td>
        <Avatar
          size={Avatar.Size.MEDIUM}
          type={Avatar.Type.EMOJI}
          backgroundColor="#D6F5E3"
          outlined
        >
          <img
            style={{ width: 30, height: 30 }}
            src="https://transferwise.com/public-resources/assets/balances/savings-emoji/google/ðŸŒ±.png"
            alt=""
          />
        </Avatar>
      </td>
    </tr>
    <tr>
      <td>Avatar.Type.EMOJI</td>
      <td>Avatar.Type.SMALL</td>
      <td>
        <Avatar size={Avatar.Size.SMALL} type={Avatar.Type.EMOJI} backgroundColor="#D6F5E3">
          <img
            style={{ width: 30, height: 30 }}
            src="https://transferwise.com/public-resources/assets/balances/savings-emoji/google/ðŸŒ±.png"
            alt=""
          />
        </Avatar>
      </td>
      <td>
        <Avatar
          size={Avatar.Size.SMALL}
          type={Avatar.Type.EMOJI}
          backgroundColor="#D6F5E3"
          outlined
        >
          <img
            style={{ width: 16, height: 16 }}
            src="https://transferwise.com/public-resources/assets/balances/savings-emoji/google/ðŸŒ±.png"
            alt=""
          />
        </Avatar>
      </td>
    </tr>
    <tr>
      <td>Avatar.Type.INITIALS</td>
      <td>Avatar.Type.LARGE</td>
      <td>
        <Avatar type={Avatar.Type.INITIALS} size={Avatar.Size.LARGE}>
          TW
        </Avatar>
      </td>
      <td>
        <Avatar type={Avatar.Type.INITIALS} size={Avatar.Size.LARGE} outlined>
          TW
        </Avatar>
      </td>
    </tr>
    <tr>
      <td>Avatar.Type.INITIALS</td>
      <td>Avatar.Type.MEDIUM</td>
      <td>
        <Avatar type={Avatar.Type.INITIALS} size={Avatar.Size.MEDIUM}>
          TW
        </Avatar>
      </td>
      <td>
        <Avatar type={Avatar.Type.INITIALS} size={Avatar.Size.MEDIUM} outlined>
          TW
        </Avatar>
      </td>
    </tr>
    <tr>
      <td>Avatar.Type.INITIALS</td>
      <td>Avatar.Type.SMALL</td>
      <td>
        <Avatar type={Avatar.Type.INITIALS} size={Avatar.Size.SMALL}>
          TW
        </Avatar>
      </td>
      <td>
        <Avatar type={Avatar.Type.INITIALS} size={Avatar.Size.SMALL} outlined>
          TW
        </Avatar>
      </td>
    </tr>
    <tr>
      <td>Avatar.Type.INITIALS</td>
      <td>Avatar.Type.MEDIUM</td>
      <td>
        <Avatar theme={Avatar.Theme.DARK} type={Avatar.Type.INITIALS} size={Avatar.Size.MEDIUM}>
          TW
        </Avatar>
      </td>
      <td>
        <Avatar theme={Avatar.Theme.DARK} type={Avatar.Type.INITIALS} size={Avatar.Size.MEDIUM} outlined>
          <Profile />
        </Avatar>
      </td>
    </tr>
    <tr>
      <td>Avatar.Type.ICON</td>
      <td>Avatar.Type.LARGE</td>
      <td>
        <Avatar type={Avatar.Type.ICON} size={Avatar.Size.LARGE}>
          <svg width="25" height="25" viewBox="0 0 25 25" fill="currentColor">
            <path d="M22.003 9.408l-10-7.405-10 7.405 1.195 1.595 8.805-6.52 8.805 6.52 1.195-1.595z"></path>
            <path d="M13 10v10h4v-7h2v7h2v2H3v-2h2v-7h2v7h4V10h2z"></path>
          </svg>
        </Avatar>
      </td>
      <td>
        <Avatar type={Avatar.Type.ICON} size={Avatar.Size.LARGE} outlined>
          <svg width="25" height="25" viewBox="0 0 25 25" fill="currentColor">
            <path d="M22.003 9.408l-10-7.405-10 7.405 1.195 1.595 8.805-6.52 8.805 6.52 1.195-1.595z"></path>
            <path d="M13 10v10h4v-7h2v7h2v2H3v-2h2v-7h2v7h4V10h2z"></path>
          </svg>
        </Avatar>
      </td>
    </tr>
    <tr>
      <td>Avatar.Type.ICON</td>
      <td>Avatar.Type.MEDIUM</td>
      <td>
        <Avatar type={Avatar.Type.ICON} size={Avatar.Size.MEDIUM}>
          <svg width="25" height="25" viewBox="0 0 25 25" fill="currentColor">
            <path d="M22.003 9.408l-10-7.405-10 7.405 1.195 1.595 8.805-6.52 8.805 6.52 1.195-1.595z"></path>
            <path d="M13 10v10h4v-7h2v7h2v2H3v-2h2v-7h2v7h4V10h2z"></path>
          </svg>
        </Avatar>
      </td>
      <td>
        <Avatar type={Avatar.Type.ICON} size={Avatar.Size.MEDIUM} outlined>
          <svg width="25" height="25" viewBox="0 0 25 25" fill="currentColor">
            <path d="M22.003 9.408l-10-7.405-10 7.405 1.195 1.595 8.805-6.52 8.805 6.52 1.195-1.595z"></path>
            <path d="M13 10v10h4v-7h2v7h2v2H3v-2h2v-7h2v7h4V10h2z"></path>
          </svg>
        </Avatar>
      </td>
    </tr>
    <tr>
      <td>Avatar.Type.ICON</td>
      <td>Avatar.Type.SMALL</td>
      <td>
        <Avatar type={Avatar.Type.ICON} size={Avatar.Size.SMALL}>
          <svg width="25" height="25" viewBox="0 0 25 25" fill="currentColor">
            <path d="M22.003 9.408l-10-7.405-10 7.405 1.195 1.595 8.805-6.52 8.805 6.52 1.195-1.595z"></path>
            <path d="M13 10v10h4v-7h2v7h2v2H3v-2h2v-7h2v7h4V10h2z"></path>
          </svg>
        </Avatar>
      </td>
      <td>
        <Avatar type={Avatar.Type.ICON} size={Avatar.Size.SMALL} outlined>
          <svg width="25" height="25" viewBox="0 0 25 25" fill="currentColor">
            <path d="M22.003 9.408l-10-7.405-10 7.405 1.195 1.595 8.805-6.52 8.805 6.52 1.195-1.595z"></path>
            <path d="M13 10v10h4v-7h2v7h2v2H3v-2h2v-7h2v7h4V10h2z"></path>
          </svg>
        </Avatar>
      </td>
    </tr>
  </tbody>
</table>

export const meta = {
  name: 'Avatar',
};
