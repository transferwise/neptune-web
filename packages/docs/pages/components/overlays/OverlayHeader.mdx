import { LiveEditorBlock, GeneratePropsTable } from '../../../utils';
import { OverlayHeader, Avatar, AvatarWrapper } from '@transferwise/components';
import { Briefcase as BriefcaseIcon, Profile as ProfileIcon } from '@transferwise/icons';
import code from '../../../liveEditorCode/overlayHeader.code';

Simple header component that displays a responsive logo an avatar and a close button.
The current and recommended size for logo is width = 138px height=24px.

<LiveEditorBlock
  code={code}
  scope={{ OverlayHeader, Avatar, BriefcaseIcon, ProfileIcon, AvatarWrapper }}
  display="vertical"
/>
<GeneratePropsTable componentName="OverlayHeader" />

export const meta = {
  name: 'OverlayHeader',
};
