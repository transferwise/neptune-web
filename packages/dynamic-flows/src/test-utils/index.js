import React from 'react';
import { mount } from 'enzyme';
import { Provider } from '@transferwise/components';
import { act } from 'react-dom/test-utils';

const getMockFetchPromise = (status, jsonFn, delay, signal) => {
  const response = {
    status,
    json: jsonFn,
  };
  return new Promise((resolve) => {
    let aborted = false;
    if (signal) {
      signal.addEventListener('abort', () => {
        aborted = true;
      });
    }
    setTimeout(() => {
      return !aborted && resolve(response);
    }, delay);
  });
};

const wait = (t) => {
  return act(() => {
    return new Promise((resolve) => setTimeout(resolve, t));
  });
};

function mountWithProviders(component, locale = 'en-GB', messages = {}) {
  return mount(<Provider i18n={{ locale, messages }}>{component}</Provider>);
}

export { mountWithProviders as mount, getMockFetchPromise, wait };
