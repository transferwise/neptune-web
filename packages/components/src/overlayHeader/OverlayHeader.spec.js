import React from 'react';
import { render } from '../test-utils';

import OverlayHeader from '.';

import Avatar from '../avatar';

describe('OverlayHeader', () => {
  const props = {
    avatar: (
      <Avatar type={Avatar.Type.INITIALS} size={Avatar.Size.MEDIUM}>
        TM
      </Avatar>
    ),
    logo: <img alt="logo_desktop" src="img_desktop" width="138" height="24" />,
    onClose: jest.fn(),
  };
  it('renders as expected', () => {
    expect(render(<OverlayHeader {...props} />).container).toMatchSnapshot();
  });

  it('renders separator only if avatar and onClose are provided', () => {
    const { container } = render(<OverlayHeader {...props} />);
    expect(container.querySelector('.separator')).toBeInTheDocument();
  });

  it("doesn't render separator if avatar or onClose are not provided", () => {
    const { container, rerender } = render(<OverlayHeader {...props} onClose={null} />);
    expect(container.querySelector('.separator')).not.toBeInTheDocument();
    rerender(<OverlayHeader {...props} avatar={null} />);
    expect(container.querySelector('.separator')).not.toBeInTheDocument();
  });
});
