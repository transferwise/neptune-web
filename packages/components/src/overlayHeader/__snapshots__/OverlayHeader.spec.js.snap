// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverlayHeader renders as expected 1`] = `
<div>
  <div
    class="np-header d-flex flex-wrap np-overlay-header p-a-3"
  >
    <div
      class="align-items-center d-flex"
    >
      <img
        alt="logo_desktop"
        height="24"
        src="img_desktop"
        width="138"
      />
    </div>
    <div
      class="align-items-center d-flex justify-content-end order-2"
    >
      <div
        class="tw-avatar tw-avatar--md tw-avatar--initials tw-avatar--light"
      >
        <div
          class="tw-avatar__content"
        >
          TM
        </div>
      </div>
      <span
        class="separator"
      />
      <button
        aria-label="Close"
        class="np-close-button close btn-link text-no-decoration np-close-button--large"
        type="button"
      >
        <span
          aria-hidden="true"
          class="tw-icon tw-icon-cross "
          role="presentation"
        >
          <svg
            fill="currentColor"
            focusable="false"
            height="24"
            width="24"
          >
            <path
              d="M12 13.414l7.293 7.293 1.414-1.414L13.414 12l7.293-7.293-1.414-1.414L12 10.586 4.707 3.293 3.293 4.707 10.586 12l-7.293 7.293 1.414 1.414L12 13.414z"
            />
          </svg>
        </span>
      </button>
    </div>
    <div
      class="align-items-center d-flex justify-content-center order-1 flex-grow-1"
    />
  </div>
</div>
`;
